### PR TITLE
Add On-Device Training Mac & iOS package installation options

### PR DESCRIPTION
**Pull request by carzh**
### Description
Added the option to install Mac & iOS packages for on-device training under the training installation options.

### Motivation and Context
Mac & iOS on-device training packages will be available soon.

live preview available [here](https://carzh.github.io/onnxruntime/)

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -631,6 +631,13 @@ var ot_validCombos = {
     
     "ot_android,ot_on_device,ot_java,ot_X64,ot_CPU":
     "Add a dependency on <a href='https://mvnrepository.com/artifact/com.microsoft.onnxruntime/onnxruntime-training-android' target='_blank'>com.microsoft.onnxruntime:onnxruntime-training-android</a> using Maven/Gradle and refer to the instructions <a href='https://onnxruntime.ai/docs/install/#install-for-on-device-training' target='_blank'>here.</a>",
+
+    "ot_mac,ot_on_device,ot_python,ot_X64,ot_CPU":
+    "pip install onnxruntime-training<br/><br/>*</b><a href='https://download.onnxruntime.ai/' target='blank'>Available versions</a>",
+
+    "ot_ios,ot_on_device,ot_objc,ot_X64,ot_CPU":
+        "Add 'onnxruntime-training-c' or 'onnxruntime-training-objc' using CocoaPods and refer to the <a href='https://onnxruntime.ai/docs/tutorials/mobile/' target='_blank'>mobile deployment guide</a>",
```
Comment (edgchen1) [inline js/script.js:639]: I think these instructions can also be enabled for C and C++ APIs

Comment (carzh) [reply js/script.js:639]: for iOS platform?

Comment (edgchen1) [reply js/script.js:639]: > for iOS platform?

yep. actually, `onnxruntime-training-objc` for Objective-C and `onnxruntime-training-c` for C/C++ would match the inference package instructions.

image
image

File: js/script.js [hunk 2]
```diff
@@ -631,6 +631,19 @@ var ot_validCombos = {
...
```
Comment (edgchen1) [inline js/script.js:639]:
```suggestion
        "Add 'onnxruntime-training-objc' using CocoaPods and refer to the <a href='https://onnxruntime.ai/docs/tutorials/mobile/' target='_blank'>mobile deployment guide</a>",
```

File: js/script.js [hunk 3]
```diff
@@ -631,6 +631,19 @@ var ot_validCombos = {
...
+    
+    "ot_ios,ot_on_device,ot_c,ot_X64,ot_CPU":
+        "Add 'onnxruntime-training-c' or 'onnxruntime-training-objc' using CocoaPods and refer to the <a href='https://onnxruntime.ai/docs/tutorials/mobile/' target='_blank'>mobile deployment guide</a>",
```
Comment (edgchen1) [inline js/script.js:642]:
```suggestion
        "Add 'onnxruntime-training-c' using CocoaPods and refer to the <a href='https://onnxruntime.ai/docs/tutorials/mobile/' target='_blank'>mobile deployment guide</a>",
```

File: js/script.js [hunk 4]
```diff
@@ -631,6 +631,19 @@ var ot_validCombos = {
...
+    
+    "ot_ios,ot_on_device,ot_cplusplus,ot_X64,ot_CPU":
+        "Add 'onnxruntime-training-c' or 'onnxruntime-training-objc' using CocoaPods and refer to the <a href='https://onnxruntime.ai/docs/tutorials/mobile/' target='_blank'>mobile deployment guide</a>",
```
Comment (edgchen1) [inline js/script.js:645]:
```suggestion
        "Add 'onnxruntime-training-c' using CocoaPods and refer to the <a href='https://onnxruntime.ai/docs/tutorials/mobile/' target='_blank'>mobile deployment guide</a>",
```

**Review by edgchen1 — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

**Comment by baijumeswani** [comment]
We now have the cpu package `onnxruntime-training-cpu` that we will publish to ADO feeds. So, we may need to update this page again.
It most likely will be 

```
pip install onnxruntime-training-cpu -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT/pypi/simple/
```

We can go ahead and merge this in and update this page again after the release.

**Pull request merged**